Multisig/UI/Settings/Passcode/PasscodeSettingsViewController.swift

```diff
@@ -80,42 +86,105 @@ class PasscodeSettingsViewController: UITableViewController {
     }
 
     private func deletePasscode() {
-        let vc = EnterPasscodeViewController()
-        let nav = UINavigationController(rootViewController: vc)
+        withPasscodeAuthentication(for: "Enter Passcode") { [unowned self] success, _, finish in
+            if success {
+                disablePasscode()
+            }
+            finish()
+        }
+    }
 
-        vc.completion = { [weak nav, unowned self] success in
+    private func disablePasscode() {
+        do {
+            try App.shared.auth.deletePasscode()
+            App.shared.snackbar.show(message: "Passcode disabled")
+        } catch {
+            let uiError = GSError.error(
+                description: "Failed to delete passcode",
+                error: GSError.GenericPasscodeError(reason: error.localizedDescription))
+            App.shared.snackbar.show(error: uiError)
+        }
+    }
+
+    private func changePasscode() {
+        withPasscodeAuthentication(for: "Change Passcode", tracking: .changePasscode) { success, nav, finish in
             if success {
+                let changeVC = ChangePasscodeEnterNewViewController {
+                    finish()
+                }
+                nav?.pushViewController(changeVC, animated: true)
+            } else {
+                finish()
+            }
+        }
+    }
+
+    private func toggleUsage(option: PasscodeOptions) {
+        withPasscodeAuthentication(for: "Require to open app") { [unowned self] success, _, finish in
```

Comment:
should this string depend on the option?

Reply:
yes, good catch